packages/file-search/src/node/file-search-service-impl.ts

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (C) 2017 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { FileSearchService } from '../common/file-search-service';
+import * as fs from "fs";
+import * as path from "path";
+import URI from '@theia/core/lib/common/uri';
+import { FileUri } from '@theia/core/lib/node';
+import * as fuzzy from 'fuzzy';
+import { injectable, inject } from 'inversify';
+import { ILogger } from '@theia/core';
+import { findContainingGitIgnore, NO_IGNORE, GitIgnore, findGitIgnore } from './git-ignore';
+
+type Filtered = (simpleName: string, fullPath: string) => boolean;
+
+
+@injectable()
+export class FileSearchServiceImpl implements FileSearchService {
+
```

Comment:
In general could all this be done async ? 
I'm concerned with for example network directories in there. If you have:

 workspace/dir
 workspace/somenetworkdir

Even if you're always searching for a file in `workspace/dir/` You will always be as slow as the `somenetworkdir`

Also given it happens to me a lot that my network drive fails and just freezes any requests to it's contents it would freeze the whole request there too.

Could you have this as a batch of results sent in a notification aync ?

Reply:
Reading more it seems quite possible to do especially given the limits we have on the fuzzy search we could not even use notifications and just resolve the find once we reach the 100/2000 limit with an async directory read.

Reply:
Or if we don't reach it keep waiting...

Reply:
I don't know what you are talking about. But maybe we should wait for real problems before making things more complex.

Reply:
I mean a blocked FS operation here will freeze the whole backend !
We can't use sync operations in the backend that's a very real problem.

Reply:
You can simulate this easily by adding a NFS/CIFS share in your workspace and killing your network connection.

The backend will become unresponsive.

Reply:
I changed it to use async methods, only. We should apply similar changes in other places.